.storybook/preview.js

```diff
@@ -50,30 +46,9 @@ export const globalTypes = {
 }
 
 export const decorators = [
-  (Story, context) => {
-    const mode = context.globals.theme
-    const theme = createTheme({ palette: { mode } })
-
-    return (
-      <EmotionThemeProvider
-        // Bug: Custom theme not propagated within Storybook.js
-        // https://github.com/mui-org/material-ui/issues/24282#issuecomment-859393395
-        theme={theme}
-      >
-        <ThemeProvider theme={theme}>
-          <CssBaseline />
-
-          <I18nProvider>
-            <RemoteConfigProvider {...remoteConfig}>
-              <CentraProvider>
-                <GlobalProvider>
-                  <Story />
-                </GlobalProvider>
-              </CentraProvider>
-            </RemoteConfigProvider>
-          </I18nProvider>
-        </ThemeProvider>
-      </EmotionThemeProvider>
-    )
-  },
+  (Story, context) => (
+    <RootProvider settings={settings} theme={context.globals.theme}>
+      <Story />
+    </RootProvider>
+  ),
```

Comment:
Hm, interesting!
Please fill me in; Have you moved the composition to a separate declaration, called `RootProvider`?